src/librawspeed/decoders/NefDecoder.cpp

```diff
@@ -115,13 +115,14 @@ RawImage NefDecoder::decodeRawInternal() {
 
   mRaw->dim = iPoint2D(width, height);
 
-  raw = mRootIFD->getIFDWithTag(static_cast<TiffTag>(0x8c));
-
   const TiffEntry* meta;
-  if (raw->hasEntry(static_cast<TiffTag>(0x96))) {
+  try {
+    raw = mRootIFD->getIFDWithTag(static_cast<TiffTag>(0x96));
     meta = raw->getEntry(static_cast<TiffTag>(0x96));
-  } else {
-    meta = raw->getEntry(static_cast<TiffTag>(0x8c)); // Fall back
+  } catch (const TiffParserException&) {
+    // Fall back
```

Comment:
Please don't do this kind of stuff.
There's `hasEntry()`.

Reply:
The problem was that the IFD couldn't be fond. I can't run `hasEntry()` if I don't have an IFD and the exception was thrown previosuly, no?

Feel free to refactor in a better way, but the point is that we need to fetch an IFD with either 0x96 or 0x8c.

Reply:
Then see what `getIFDWithTag()` does, and call `getIFDsWithTag()` directly,
and fall back to the other tag if `getIFDsWithTag()` returned nothing.
My point: don't just catch exceptions if the code can be written to handle it more gracefully.

Reply:
Done.

Reply:
I guess one could go a step further and remove `hasEntry()` redundancy, it should be implied by `getIFDsWithTag()`/`getIFDWithTag()`...

Reply:
Or skip looking for the IFD altogether as it's not used, we just need the tag entry.